snapcraft/tests/fake_servers/__init__.py

```diff
@@ -103,6 +103,15 @@ def do_GET(self):
                     ('maintainer', 'none'),
                 ))),
             ))
+            if 'CUSTOM_PARTS' in os.environ:
```

Comment:
Why not use an `if`/`else` here instead of overwriting the response we just made?